BLE_SupportedFeatures/source/main.cpp

```diff
@@ -0,0 +1,164 @@
+        printf("STABLE_MODULATION_INDEX_TRANSMITTER feature supported %d\n", _ble.gap().isFeatureSupported(ble::controller_supported_features_t::STABLE_MODULATION_INDEX_TRANSMITTER));
+        printf("STABLE_MODULATION_INDEX_RECEIVER feature supported %d\n", _ble.gap().isFeatureSupported(ble::controller_supported_features_t::STABLE_MODULATION_INDEX_RECEIVER));
+        printf("LE_CODED_PHY feature supported %d\n", _ble.gap().isFeatureSupported(ble::controller_supported_features_t::LE_CODED_PHY));
+        printf("LE_EXTENDED_ADVERTISING feature supported %d\n", _ble.gap().isFeatureSupported(ble::controller_supported_features_t::LE_EXTENDED_ADVERTISING));
+        printf("LE_PERIODIC_ADVERTISING feature supported %d\n", _ble.gap().isFeatureSupported(ble::controller_supported_features_t::LE_PERIODIC_ADVERTISING));
+        printf("CHANNEL_SELECTION_ALGORITHM_2 feature supported %d\n", _ble.gap().isFeatureSupported(ble::controller_supported_features_t::CHANNEL_SELECTION_ALGORITHM_2));
+        printf("LE_POWER_CLASS feature supported %d\n", _ble.gap().isFeatureSupported(ble::controller_supported_features_t::LE_POWER_CLASS));
+
+        /* this allows us to capture events like onConnectionComplete() */
+        _ble.gap().setEventHandler(this);
+
+        start_advertising();
+    }
+
+    void start_advertising()
+    {
+        /* Configure advertising parameters */
+
+        ble::AdvertisingParameters adv_parameters(
+            ble::advertising_type_t::NON_CONNECTABLE_UNDIRECTED,
+            ble::adv_interval_t(ble::millisecond_t(100))
+        );
+
+
+        /* Configure advertising payload */
+
+        _adv_data_builder.setFlags();
+        _adv_data_builder.setAppearance(ble::adv_data_appearance_t::UNKNOWN);
+        _adv_data_builder.setLocalServiceList({&_service_uuid, 1});
+        _adv_data_builder.setName(DEVICE_NAME);
+
+
+        /* Set advertising parameters configuration to our BLE instance */
+
+        ble_error_t error = _ble.gap().setAdvertisingParameters(
+            ble::LEGACY_ADVERTISING_HANDLE,
+            adv_parameters
+        );
+
+        if (error) {
+            printf("_ble.gap().setAdvertisingParameters() failed\r\n");
+            return;
+        }
+
+        /* Set advertising payload configuration to our BLE instance */
+
+        error = _ble.gap().setAdvertisingPayload(
+            ble::LEGACY_ADVERTISING_HANDLE,
+            _adv_data_builder.getAdvertisingData()
+        );
+
+        if (error) {
+            printf("_ble.gap().setAdvertisingPayload() failed\r\n");
+            return;
+        }
+
+        /* Start advertising */
+
+        error = _ble.gap().startAdvertising(ble::LEGACY_ADVERTISING_HANDLE);
+
+        if (error) {
+            printf("_ble.gap().startAdvertising() failed\r\n");
+            return;
+        }
+
+        printf("Device is advertising\r\n");
+
+    }
```

Comment:
this is a duplicate of the dedicated advertising demo, I think we can skip this

Reply:
Yes, i started from that example, not from scratch.